benchmarks/experiment_runner.py

```diff
@@ -262,59 +263,56 @@ def run_single_config(self):
-                                                      experiment_config,
-                                                      model_config,
-                                                      repeat_iteration)
-        for k, v in metrics.items():
-          if k not in accumulated_metrics:
-            accumulated_metrics[k] = []
-          accumulated_metrics[k].append(v)
-
-    # Save the dict representation before deleting them.
-    # This will be used later for saving the results.
-    experiment_dict = benchmark_experiment.to_dict()
-    model_dict = benchmark_model.to_dict()
-
-    # Save other model-specific configuration: tolerance and whether
-    # to use cosine similarity on accuracy checks.
-    tolerance = benchmark_model.tolerance()
-    use_cosine_similarity = benchmark_model.use_cosine_similarity()
-    skip_verifier = benchmark_model.skip_verifier()
-
-    # Delete the instantiated BenchmarkModel, so we can save memory
-    # for verifying the result.
-    del benchmark_model
-    cleanup(benchmark_experiment.is_cuda())
-
-    # Run the verifier iff:
-    #
-    #   1. We are running this script with --verify flag
-    #   2. It should not be skipped
-    if self._args.verify and not skip_verifier:
-      res = verify(self, experiment_config, model_config, tolerance,
-                   use_cosine_similarity)
-    else:
-      res = VerificationCode.VERIFIER_SKIPPED
-
-    self._save_results(experiment_dict, model_dict, accumulated_metrics, res)
+    # Only run the actual experiment first if the --verify flag is not
+    # specified. This is so we avoid using too much memory before running
+    # eager.
+    if not self._args.verify:
+      reset_rng_state(experiment)
+      model = self.model_loader.load_model(model_config, experiment)
+
+      # Repeat the experiment and accumulate metrics.
+      with model.pick_grad():
+        for repeat_iteration in range(self._args.repeat):
+          metrics, _ = self.run_once_and_gather_metrics(experiment, model,
+                                                        experiment_config,
+                                                        model_config,
+                                                        repeat_iteration)
+          for k, v in metrics.items():
+            if k not in accumulated_metrics:
+              accumulated_metrics[k] = []
+            accumulated_metrics[k].append(v)
+    elif not model.skip_verifier():
+      try:
+        verification_code = verify(
+            self,
+            experiment_config,
+            model_config,
+            tolerance=model.tolerance(),
+            use_cosine_similarity=model.use_cosine_similarity())
+      except VerificationException as e:
+        verification_code = e.code
+        # Record the error in the metrics dictionary.
+        # Similar to what's done when the whole experiment fails.
+        accumulated_metrics["error"] = traceback.format_exc()
```

Comment:
If we use `verify`, we will not capture any performance metrics expect the verify result, did I understand it correctly?

Reply:
Yes. That's correct.